src/destination/destinationHandlers/smtpHandler.ts

```diff
@@ -49,74 +56,107 @@ export class SMTPService implements Handler {
         return true
     }
 
+    private preparePaylodAndSend(event: Event, smtpTemplate: NotificationTemplates, setting: NotificationSettings, p: string){
+          let sdk: NotifmeSdk = new NotifmeSdk({
+                channels: {
+                    email: {
+                        providers: [{
+                           type: 'smtp',
+                           port: this.smtpConfig['port'],
+                           host: this.smtpConfig['host'],
+                           auth:{
+                            user: this.smtpConfig['auth_user'],
+                            pass: this.smtpConfig['auth_password'],
+                           }
+                        }]
+                    }
+                }
+            });
+          event.payload['fromEmail'] = this.smtpConfig['from_email']
+          let engine = new Engine();
+          // let options = { allowUndefinedFacts: true }
+          let conditions: string = p['rule']['conditions'];
+
+          this.logger.info('smtpConfigRepository.preparePaylodAndSend()')
+          //this.logger.info(JSON.stringify(event))
+          if (conditions) {
+              engine.addRule({conditions: conditions, event: event});
+              engine.run(event).then(e => {
+                  this.sendNotification(event, sdk, smtpTemplate.template_payload).then(result => {
+                      this.saveNotificationEventSuccessLog(result, event, p, setting);
+                  }).catch((error) => {
+                      this.logger.error(error.message);
+                      this.saveNotificationEventFailureLog(event, p, setting);
+                  });
+              })
+          } else {
+              this.sendNotification(event, sdk, smtpTemplate.template_payload).then(result => {
+                  this.saveNotificationEventSuccessLog(result, event, p, setting);
+              }).catch((error) => {
+                  this.logger.error(error.message);
+                  this.saveNotificationEventFailureLog(event, p, setting);
+              });
+          }
+    }
+
     private processNotification(userId: number, event: Event, smtpTemplate: NotificationTemplates, setting: NotificationSettings, p: string, emailMap: Map<string, boolean>) {
         this.usersRepository.findByUserId(userId).then(user => {
             if (!user) {
                 this.logger.info('no user found for id')
                 this.logger.info(event.correlationId)
                 return
             }
+            this.logger.info('usersRepository.findByUserId')
```

Comment:
can we remove or use debug for unnecessary logs for prod

Reply:
For testing only, will remove before merging